### PR TITLE
[feat] introduce new remote user store impl - add file verification steps

### PR DESCRIPTION
**Pull request by JayaShakthi97**
### Purpose

$subject

### Related Issues

- N/A

### Related PRs

- N/A

### Checklist

- [ ] e2e cypress tests locally verified. (for internal contributers)
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Relevant backend changes deployed and verified
- [ ] [Unit tests](/docs/testing/UNIT_TESTING.md) provided. (Add links if there are any)
- [ ] [Integration tests](https://github.com/wso2/product-is/tree/master/modules/integration) provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines
- [ ] Ran FindSecurityBugs plugin and verified report
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets

### CODE REVIEW

File: features/admin.remote-userstores.v1/components/edit/setup-guide/remote/download-step.tsx [hunk 1]
```diff
@@ -50,52 +75,168 @@ const DownloadAgentStep: FunctionComponent<DownloadAgentStepPropsInterface> = ({
...
+    useEffect(() => {
+        if (checkSumFetchRequestError) {
+            dispatch(addAlert({
+                description: t("remoteUserStores:notifications.checkSumError.description"),
+                level: AlertLevels.SUCCESS,
+                message: t("remoteUserStores:notifications.checkSumError.message")
+            }));
+        }
+    }, [ checkSumFetchRequestError ]);
+
```
Comment (JayaShakthi97) [inline features/admin.remote-userstores.v1/components/edit/setup-guide/remote/download-step.tsx:134]:
```suggestion
```

File: features/admin.remote-userstores.v1/components/edit/setup-guide/remote/download-step.tsx [hunk 2]
```diff
@@ -50,52 +75,168 @@ const DownloadAgentStep: FunctionComponent<DownloadAgentStepPropsInterface> = ({
...
+                { availableOptions.map((option: (AgentDownloadInfoInterface & { os: OperatingSystem })) => {
                     return (
                         <Button
                             key={ option.os }
                             data-componentid={ `${componentId}-${option.os}-button` }
                             size="small"
                             variant="outlined"
                             startIcon={ <DownloadIcon /> }
-                            onClick={ () => onDownloadClicked(option.link) }
+                            onClick={ () => onDownloadClicked(option.os,option.file) }
```
Comment (JayaShakthi97) [inline features/admin.remote-userstores.v1/components/edit/setup-guide/remote/download-step.tsx:183]:
```suggestion
                            onClick={ () => onDownloadClicked(option.os, option.file) }
```

File: features/admin.remote-userstores.v1/components/edit/setup-guide/remote/download-step.tsx [hunk 3]
```diff
@@ -50,52 +84,168 @@ const DownloadAgentStep: FunctionComponent<DownloadAgentStepPropsInterface> = ({
     ["data-componentid"]: componentId = "download-agent-step"
 }: DownloadAgentStepPropsInterface): ReactElement => {
     const { t } = useTranslation();
+    const dispatch: Dispatch = useDispatch();
+
+    const [ downloadedOS, setDownloadedOS ] = useState<OperatingSystem>();
 
-    const availableOptions: { os: string; link: string }[] = [
+    const availableOptions: (AgentDownloadInfoInterface & { os: OperatingSystem })[] = [
```
[the remark this answers is not in the document]
Comment (JayaShakthi97) [reply features/admin.remote-userstores.v1/components/edit/setup-guide/remote/download-step.tsx]: downloadURLs prop won't contain the `os` attribute. Defined a new interface 877e7af2560d27616792184a78d3c85febf4a635

File: features/admin.remote-userstores.v1/api/use-get-sha-file.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+/**
+ * Hook to get DataType.
+ *
+ * @param filePath - File path to be fetched.
+ * @param shouldFetch - If true, will fetch the data.
+ *
+ * @returns content of the file.
+ */
+const useGetCheckSum = <Data = string, Error = RequestErrorInterface>(
```
[the remark this answers is not in the document]
Comment (JayaShakthi97) [reply features/admin.remote-userstores.v1/api/use-get-sha-file.ts:35]: Updated 877e7af2560d27616792184a78d3c85febf4a635

File: features/admin.remote-userstores.v1/components/edit/setup-guide/remote/download-step.tsx [hunk 5]
```diff
@@ -50,52 +94,168 @@ const DownloadAgentStep: FunctionComponent<DownloadAgentStepPropsInterface> = ({
...
+    useEffect(() => {
+        if (checkSumFetchRequestError) {
+            dispatch(addAlert({
+                description: t("remoteUserStores:notifications.checkSumError.description"),
+                level: AlertLevels.SUCCESS,
+                message: t("remoteUserStores:notifications.checkSumError.message")
+            }));
+        }
+    }, [ checkSumFetchRequestError ]);
+
```
Comment (pavinduLakshan) [inline features/admin.remote-userstores.v1/components/edit/setup-guide/remote/download-step.tsx:153]:
```suggestion
```

File: features/admin.remote-userstores.v1/components/edit/setup-guide/remote/download-step.tsx [hunk 6]
```diff
@@ -50,52 +94,167 @@ const DownloadAgentStep: FunctionComponent<DownloadAgentStepPropsInterface> = ({
...
+                { availableOptions.map((option: AgentDownloadOptionInterface) => {
                     return (
                         <Button
                             key={ option.os }
                             data-componentid={ `${componentId}-${option.os}-button` }
                             size="small"
                             variant="outlined"
                             startIcon={ <DownloadIcon /> }
-                            onClick={ () => onDownloadClicked(option.link) }
+                            onClick={ () => onDownloadClicked(option.os,option.file) }
```
Comment (JayaShakthi97) [inline features/admin.remote-userstores.v1/components/edit/setup-guide/remote/download-step.tsx:201]:
```suggestion
                            onClick={ () => onDownloadClicked(option.os, option.file) }
```

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**